mockwebserver/src/main/java/com/squareup/okhttp/mockwebserver/MockWebServer.java

```diff
@@ -565,7 +585,83 @@ private RecordedRequest readRequest(Socket socket, InputStream in, OutputStream
         new Buffer().write(requestBody.toByteArray()), sequenceNumber, socket);
   }
 
-  private void writeResponse(Socket socket, OutputStream out, MockResponse response)
+  private void handleWebSocketUpgrade(Socket socket, InputStream in, OutputStream out,
+      RecordedRequest request, MockResponse response) throws IOException {
+    response.setHeader("Connection", "Upgrade");
```

Comment:
I think we should do this work in `MockResponse.withWebSocketUpgrade`. That way we can test scenarios that later remove those headers.